doc/ref_arch/kubernetes/chapters/chapter03.md

```diff
@@ -238,7 +238,11 @@ Non-resilient applications are sensitive to platform impairments on Compute like
 <p align="center"><b>Table 3-1:</b> Categories of applications, requirements for scheduling pods and Kubernetes features</p>
 
 Kubernetes clusters using above enhancements can implement worker nodes with "bare metal" servers (running Container Runtime in Linux host Operating System) or with virtual machines (VMs, on hypervisor).
-When running in VMs, the following requirements and configuration are needed for performance optimization: CPU Manager for vCPUs in VM. Huge pages enabled in hypervisor and mapped into VM. Hardware Topology Management with hypervisors enabled for NUMA. Node Feature Discovery and Device Plugin Framework for CPU instructions and devices exposed through hypervisor.
+When running in VMs, the following list of configurations shows what is needed for non-resilient applications:
+* CPU Manager managing vCPUs that hypervisor provides into VM.
+* Huge pages enabled in hypervisor, mapped into VM, enabled in guest OS, and mapped into pod.
+* Hardware Topology Management with NUMA enabled in hypervisor, mapped into VM, if needed enabled in guest OS, and mapped into pod.
+* Node Feature Discovery and Device Plugin Framework for CPU instructions and devices enabled in hypervisor, mapped into VM, enabled in guest OS, and mapped into pod.
```

Comment:
Aren't Node Feature Discovery or Device Plugins independent of the Guest OS? So, they can't be enabled in the Guest OS. Correct?

Reply:
> Aren't Node Feature Discovery or Device Plugins independent of the Guest OS? So, they can't be enabled in the Guest OS. Correct?

@pgoyal01 : After studying more documentation and thinking about recent projects, with commit [25ac683](https://github.com/cntt-n/CNTT/pull/2417/commits/25ac683a4210b1a8cf86f892cd00458d4f6f2b9e) I added "if needed". For NFD it can depend on kernel module and kernel configuration. For Device Plugin it can require Linux kernel driver. CPU instructions and host-device CNI will work.